### PR TITLE
to close #138 detect if map move depends on gestures

### LINKED ISSUE #138: Gesture detection, a way to detect user gesture
i need to locate when user move the map, or when the map is moved by the controller.
So "onPositionChanged" is called in both situation.

The case is this:
The map is updated on location change, so i use controller to change the center replacing the user location. this is a basic navigation app..
now i need to stop map update only if user shift/touch the map.
Both resulting in onPositionChanged, there is a way to detect it?

### PR DESCRIPTION
**Pull request by gimox**
Added hasGesture property on MapPosition. OnPositionChanged user can check if map is moved by controller or by user gestures.

### CODE REVIEW

**Comment by vinicentus** [comment]
+1

File: lib/flutter_map.dart [hunk 1]
```diff
@@ -136,5 +136,6 @@ class MapPosition {
   final LatLng center;
   final LatLngBounds bounds;
   final double zoom;
-  MapPosition({this.center, this.bounds, this.zoom});
+  final bool hasGesture;
+  MapPosition({this.center, this.bounds, this.zoom, this.hasGesture = false});
```
Comment (johnpryan) [inline lib/flutter_map.dart]: a MapPosition is not an event, so I would prefer that the `onPositionChanged` stream emits a new object; `MapPositionChangedEvent` with two fields; the MapPosition and a bool representing whether or not the map has moved due to the current user.

Comment (gimox) [reply lib/flutter_map.dart]: ok now MapPositionChangedEvent emit two events

**Comment by gimox** [comment]
new usage:

```
 onPositionChanged: (MapPosition mapPosition, bool hasGesture) {
                    if(hasGesture == true) {
                     ...
                    }
                  },
```

File: lib/src/map/map.dart [hunk 2]
```diff
@@ -20,8 +20,8 @@ class MapControllerImpl implements MapController {
     }
   }
 
-  void move(LatLng center, double zoom) {
-    _state.move(center, zoom);
+  void move(LatLng center, double zoom, [bool hasGesture = false]) {
```
Comment (johnpryan) [inline lib/src/map/map.dart]: consider making this a named optional rather than an unnamed optional param

Comment (gimox) [reply lib/src/map/map.dart]: why?

Comment (johnpryan) [reply lib/src/map/map.dart]: just something to consider for readability

**Comment by johnpryan** [comment]
Looks good, thanks for your patience!

**Comment by AlexandreRoba** [comment]
This seems to be broken on version 0.5.0

**Pull request merged**